feat(apps/whale-api): add new endpoint to expose mint limits and minted amount per member

lykalabrada opened the pull request:
<!--  Thanks for sending a pull request! -->

#### What this PR does / why we need it:
As described on the linked issue below

#### Which issue(s) does this PR fixes?:
<!--
(Optional) Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes https://github.com/JellyfishSDK/jellyfish/issues/1961

#### Additional comments?:

lykalabrada committed:
feat(apps/whale-api): add new endpoint to expose mint limits and minted amounts per member

lykalabrada committed:
feat(apps/whale-api): fix consortium test

lykalabrada committed:
feat(apps/whale-api): add next-day scenario to mint stats api

lykalabrada committed:
Merge branch 'main' of lykalabrada/jellyfish into lyka/member-mint-stats-api

lykalabrada committed:
feat(apps/whale-api): refactor getMemberMintStats function

lykalabrada committed:
Merge branch 'main' into lyka/member-mint-stats-api

lykalabrada committed:
feat(apps/whale-api): update blocks to generate to trigger next day

lykalabrada committed:
Merge branch 'main' into lyka/member-mint-stats-api

lykalabrada committed:
Merge branch 'main' into lyka/member-mint-stats-api

lykalabrada committed:
feat(apps/whale-api): get mint limits - update mintTokens params

lykalabrada committed:
feat(apps/whale-api): remove commented code

pierregee reviewed (changes requested):
We still need the global limit somewhere in the response. One scenario that could happen is that there's a global limit but no member limit

lykalabrada committed:
Merge branch 'main' into lyka/member-mint-stats-api

pierregee commented:
Need to cover [tests](https://github.com/JellyfishSDK/jellyfish/blob/main/packages/whale-api-client/__tests__/api/consortium.test.ts) on whale-api-client for this new endpoint too

lykalabrada committed:
chore(apps/whale-api): add test for mint stats api on whale-client

lykalabrada committed:
feat(apps/whale-api): get mint limits - add global mint stats

lykalabrada commented:
> We still need the global limit somewhere in the response. One scenario that could happen is that there's a global limit but no member limit

Added global mint stats. Also added the test on the `whale-api-client` side

lykalabrada committed:
feat(apps/whale-api): get mint limits - update some types

pierregee commented on apps/whale-api/src/module.api/consortium.controller.spec.ts:
```diff
@@ -3,125 +3,126 @@ import { TestingGroup } from '@defichain/jellyfish-testing'
+}
+
+async function setMemberInfo (tokenId: string, memberInfo: Array<{ id: string, name: string, backingId: string, ownerAddress: string, mintLimit: string, dailyMintLimit: string }>): Promise<void> {
+  const members: any = {}
```
all  good, just one nitpick , this is better? 
```suggestion
const members: { [key: string]: { [key: string]: string } } = {}
```

lykalabrada committed:
chore(test): update consortium member type

lykalabrada committed:
Merge branch 'main' into lyka/member-mint-stats-api

lykalabrada committed:
Merge branch 'main' into lyka/member-mint-stats-api

lykalabrada committed:
Merge branch 'main' into lyka/member-mint-stats-api

thedoublejay commented on apps/whale-api/src/module.api/consortium.controller.ts:
```diff
@@ -23,4 +23,18 @@ export class ConsortiumController {
+   *
+   * @return {Promise<MemberMintStatsInfo>}
+    */
+  @Get('/mintstats/:memberid')
```
Would this be better to just be stats? So it can contain mint and burn stats in the future?

lykalabrada replied:
Updated!

lykalabrada committed:
Merge branch 'main' into lyka/member-mint-stats-api

lykalabrada committed:
feat(apps/whale-api): rename member stats api to be generic
